WalletWasabi/WabiSabi/Crypto/CredentialRequesting/ImmutableValueSequence`1.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Crypto.CredentialRequesting
+{
+	/// <summary>
+	/// ImmutableValueSequence represents an immutable sequence with value semantics, meaning that
+	/// it can be compared with other sequences and the equality of two sequences is based on the
+	/// equality of the elements.
+	/// </summary>
+	public class ImmutableValueSequence<T> : IEnumerable<T>, IEquatable<IEnumerable<T>> where T : IEquatable<T>
```

Comment:
Isn't it more conservative to only implement `IEquatable<ImmutableValueSequence<T>>`?

For example, I think this hypothetical test might fail, and doesn't make sense to me anyway:

```cs
IEnumerable<T> a = ...;
ImmutableValueSequence<T> = ...;

Assert.Equal(a == b, b == a);
```